src/protocol.rs

```diff
@@ -657,7 +659,22 @@ impl Mortal for SessionPtr {
     }
 
     fn retire_at(&self, srv: &CryptoServer) -> Option<Timing> {
-        self.created_at(srv).map(|t| t + REKEY_AFTER_TIME)
+        // If we where the initiator, wait an extra ten seconds to avoid
+        // both parties starting the handshake at the same time. In most situations
+        // this should provide ample time for the other party to perform a
+        // complete handshake before this peer starts another handshake.
+        // This also has the peers going back and forth taking the initiator role
+        // and responder role.
+        use HandshakeRole::*;
+        let role = self.get(srv).as_ref().map(|p| p.handshake_role);
+        let extra_wait = match role {
+            Some(Initiator) => 10.0,
+            Some(Responder) => 0.0,
+            None => 0.0,
+        };
+
+        self.created_at(srv)
+            .map(|t| t + REKEY_AFTER_TIME + extra_wait)
```

Comment:
If we "change" the value of `REKEY_AFTER_TIME`, wouldn't it make sense to make this constant private and change its name then? From a software engineering point of view, constants imply to me somewhat of an absolute truth, which wouldn't be true with this patch anymore.

I would suggest changing this constant's name to something like "MIN_REKEY_AFTER_TIME" and maybe making it private.

Reply:
Thats a bit of a bike shed, but ok :shrug: